entity-service/src/main/kotlin/com/egm/stellio/entity/repository/QueryUtils.kt

```diff
@@ -229,9 +229,10 @@ object QueryUtils {
             """.trimIndent()
         else
             """
-                WITH collect(distinct(entity.id)) as entityIds, count(entity.id) as count
+                WITH collect(distinct(entity.id)) as entityIds, collect(DISTINCT(labels(entity))) as entityTypes, count(entity.id) as count
```

Comment:
I think that you will loose the type for each entity: you'll end up with a list of entity ids, another one with entity types, but no way to know the type of each entity

Reply:
I'm wondering if not easier to just collect the `entity` (since in this case, you want a set of info from the entity, and not just the id)

Reply:
I have problem to collect only the entity because database return entity with `labels` and `Entity` need a `type`.

Reply:
This must be my way to cast. I try to do `it["entities"] as Entity`. So I don't see how to cast the result of the query to an Entity

Reply:
Sorry I forgot to say than I replace `WITH collect(distinct(entity.id)) as entityIds, collect(DISTINCT(labels(entity))) as entityTypes, count(entity.id) as count` by `WITH collect(entity) as entities, count(entity.id) as count`

Reply:
so one option is to create "manually" this `Entity` object, based on what you get from each entry in `entities`

Reply:
also, don't forget you also want to get the right the user has on the entity.